### ISSUE TITLE
Change Tuple v1 to fst and v2 to snd

### ISSUE THREAD

**Issue by abueide** [jOOQ/jOOL#364]
In my opinion fst and snd is much more legible than v1 and v2, as well as being a bit easier to type without reaching for number keys. It would be nice if we could change it.

For example this code looks pretty illegible:
```
private boolean match(Tuple2<Entity, Aspect> e1, Tuple2<Entity, Aspect> e2){
        if(e1.v2.isInterested(e1.v1) && e2.v2.isInterested(e2.v1)){
            return  true;
        }else {
            return false;
        }
    }
```
This looks a bit better, but maybe its just because its clashing with my badly named e1 and e2 vs the v1 and v2.
```
private boolean match(Tuple2<Entity, Aspect> e1, Tuple2<Entity, Aspect> e2){
        if(e1.fst.isInterested(e1.fst) && e2.snd.isInterested(e2.snd)){
            return  true;
        }else {
            return false;
        }
    }
```

I'd be willing to write the PR if you guys think this is a good idea, but if a breaking change for a small style change is not in your interest just let me know.

**Comment by abueide** [comment]
I just realized how stupid this was remembering that more tuples exist than Tuple2 and that would break the consistency, sorry!

**Comment by lukaseder** [comment]
Thanks for your suggestion, @abueide. When I created jOOλ, I had briefly considered what this library did: https://www.javatuples.org

```
Unit<A> (1 element)
Pair<A,B> (2 elements)
Triplet<A,B,C> (3 elements)
Quartet<A,B,C,D> (4 elements)
```

Interestingly enough, they named their tuples without indicating the degree as a number, but then still used (zero based, yuck) accessors, like `getValue3()`, and, probably as a compromise, didn't number their generic type variables, but named them `<A, B, C, D>`.

This just to show you there isn't really a very satisfying solution that fits everyone. Much better than tuples in general would be records (with named attributes). If Java had anonymous types like C#, then we would hardly use tuples at all, which are impractical, regardless if the value is called `v2` or `second` or whatever. Because a much bettern name for it would be `lastName` (for example).

As long as type inference works (with lambdas or `var`), it would actually be possible to write things like:

```java
Stream.of("a", "b")
    .map(s -> new Object() { String name = s; })
    .forEach(o -> System.out.println(o.name));
```

But we cannot pass around this type between methods, so the utility is almost zero. We did get first class records with Java 14, though, so I have high hopes that the name of tuple attributes will become irrelevant in the future :)

Cheers,
Lukas
